sources/hyper-proxy-shim/Cargo.toml

```diff
@@ -11,39 +11,26 @@ readme = "README.md"
 keywords = ["hyper", "proxy", "tokio", "ssl"]
 categories = ["web-programming::http-client", "asynchronous", "authentication"]
 license = "MIT"
-edition = "2018"
+edition = "2021"
 
 [dependencies]
 tokio = { version = "1", features = ["io-std", "io-util"] }
 hyper = { version = "0.14", features = ["client"] }
-
-tower-service = "0.3"
 http = "0.2"
 futures-util = { version = "0.3", default-features = false }
-bytes = "1.0"
-hyper-tls = { version = "0.5.0", optional = true }
-tokio-native-tls = { version = "0.3.0", optional = true }
-native-tls = { version = "0.2", optional = true }
-openssl = { version = "0.10", optional = true }
-tokio-openssl = { version = "0.6", optional = true }
-tokio-rustls = { version = "0.22", optional = true }
-hyper-rustls = { version = "0.22", optional = true }
-
-webpki = { version = "0.21", optional = true }
-rustls-native-certs = { version = "0.5.0", optional = true }
-webpki-roots = { version = "0.21.0", optional = true }
+bytes = "1"
+tokio-rustls = { version = "0.23", optional = true }
+hyper-rustls = { version = "0.23", optional = true }
 headers = "0.3"
 
 [dev-dependencies]
-tokio = { version = "1.0", features = ["full"] }
+tokio = { version = "1", features = ["full"] }
 hyper = { version = "0.14", features = ["client", "http1", "tcp"] }
 
 [features]
-openssl-tls = ["openssl", "tokio-openssl"]
-tls = ["tokio-native-tls", "hyper-tls", "native-tls"]
 # note that `rustls-base` is not a valid feature on its own - it will configure rustls without root
 # certificates!
-rustls-base = ["tokio-rustls", "hyper-rustls", "webpki"]
-rustls = ["rustls-base", "rustls-native-certs", "hyper-rustls/native-tokio"]
-rustls-webpki = ["rustls-base", "webpki-roots", "hyper-rustls/webpki-tokio"]
-default = ["tls"]
+rustls-base = ["tokio-rustls", "hyper-rustls"]
+rustls = ["rustls-base", "hyper-rustls/native-tokio"]
+rustls-webpki = ["rustls-base", "hyper-rustls/webpki-tokio"]
+default = ["rustls"]
```

Comment:
I removed the `tls` feature to keep this from filling the sources lockfile with openssl stuff. Is there something I can do differently to avoid removing that code?